system/Database/MySQLi/Forge.php

```diff
@@ -116,11 +116,11 @@ protected function _createTableAttributes(array $attributes): string
             }
         }
 
-        if ($this->db->charset !== '' && ! strpos($sql, 'CHARACTER SET') && ! strpos($sql, 'CHARSET')) {
+        if ($this->db->charset !== '' && in_array(strpos($sql, 'CHARACTER SET'), [0, false], true) && in_array(strpos($sql, 'CHARSET'), [0, false], true)) {
```

Comment:
why is this not changed to just str_contains?

Reply:
strpos returns can be 0 or false, at `0`, it means found at at very first, while `str_contains()` return false when not found.

Reply:
If we translate the previous behavior 1 to 1 you're absolutely right. But in this case, we can really simplify things, because the goal is to not add `CHARACTER SET` if it's already present in the string.

Reply:
ok, updated https://github.com/codeigniter4/CodeIgniter4/pull/9339/commits/10f3c7e00a335f2eef2df3ff75e0b8bf704e97d6